src/main/java/com/snowflake/kafka/connector/internal/streaming/SnowflakeSinkServiceV2.java

```diff
@@ -0,0 +1,440 @@
+    if (partitionsToChannel.containsKey(partitionChannelKey)) {
+      return partitionsToChannel.get(partitionChannelKey).getCommittedOffset();
+    } else {
+      LOGGER.warn(
+          "Topic: {} Partition: {} hasn't been initialized to get offset",
+          topicPartition.topic(),
+          topicPartition.partition());
+      return 0;
+    }
+  }
+
+  @Override
+  public int getPartitionCount() {
+    return partitionsToChannel.size();
+  }
+
+  @Override
+  public void callAllGetOffset() {
+    // undefined
+  }
+
+  @Override
+  public void closeAll() {
+    partitionsToChannel.forEach(
+        (partitionChannelKey, topicPartitionChannel) -> {
+          LOGGER.info("Closing partition channel:{}", partitionChannelKey);
+          topicPartitionChannel.closeChannel();
+        });
+    partitionsToChannel.clear();
+    closeStreamingClient();
+  }
+
+  @Override
+  public void close(Collection<TopicPartition> partitions) {
+    partitions.forEach(
+        topicPartition -> {
+          String partitionChannelKey =
+              partitionChannelKey(topicPartition.topic(), topicPartition.partition());
+          LOGGER.info("Closing partition channel:{}", partitionChannelKey);
+          partitionsToChannel.get(partitionChannelKey).closeChannel();
+          partitionsToChannel.remove(partitionChannelKey);
+        });
+
+    closeStreamingClient();
+  }
+
+  @Override
+  public void setIsStoppedToTrue() {}
+
+  /* Undefined */
+  @Override
+  public boolean isClosed() {
+    return false;
+  }
+
+  @Override
+  public void setRecordNumber(long num) {
+    if (num < 0) {
+      LOGGER.error("number of record in each file is {}, it is negative, reset to 0", num);
+      this.recordNum = 0;
+    } else {
+      this.recordNum = num;
+      LOGGER.info("set number of record limitation to {}", num);
+    }
+  }
+
+  /**
+   * Assume this is buffer size in bytes, since this is streaming ingestion
+   *
+   * <p>Copying this from SinkServiceV1 and we will get away from this in future
+   *
+   * @param size a non negative long number represents data size limitation
+   */
+  @Override
+  public void setFileSize(long size) {
+    if (size < SnowflakeSinkConnectorConfig.BUFFER_SIZE_BYTES_MIN) {
+      LOGGER.error(
+          "Buffer size is {} bytes, it is smaller than the minimum buffer "
+              + "size {} bytes, reset to the default file size",
+          size,
+          SnowflakeSinkConnectorConfig.BUFFER_SIZE_BYTES_DEFAULT);
+      this.fileSizeBytes = SnowflakeSinkConnectorConfig.BUFFER_SIZE_BYTES_DEFAULT;
+    } else {
+      this.fileSizeBytes = size;
+      LOGGER.info("set buffer size limitation to {} bytes", size);
+    }
+  }
+
+  @Override
+  public void setTopic2TableMap(Map<String, String> topicToTableMap) {
+    this.topicToTableMap = topicToTableMap;
+  }
+
+  @Override
+  public void setFlushTime(long time) {
+    if (time < SnowflakeSinkConnectorConfig.BUFFER_FLUSH_TIME_SEC_MIN) {
```

Comment:
I think we will need a separate min for SS? And we will need to update the flush interval logic if we're updating this to something like 1s, probably need a BG to check it?